docs/performance/sharding.md

```diff
@@ -1,41 +1,138 @@
-(sharding-under-allocation)=
+General principles require careful consideration of cluster
+sizing and architecture.
+Keep the following things in mind when building your sharding strategy.
+Each shard incurs overhead in terms of open files, RAM allocation, and CPU cycles
+for maintenance operations.
+
+### Shard size vs. number of shards
+
+The optimal approach balances shard count with shard size. Individual shards should
+typically contain 10-50 GB of data, being the sweet spot for most
+workloads. In large clusters, this often means fewer shards than total CPU cores,
+as larger shards can still be processed efficiently by multiple CPU cores during
+query execution.
+
+### CPU-to-shard ratio
+
+If most nodes have more shards per table than they have CPUs, the cluster can
+experience performance degradations.
+For example, on clusters with substantial CPU resources (e.g., 8 nodes × 32 CPUs
+= 256 total CPUs), creating 256+ shards per table often proves counterproductive.
+If you don't manually set the number of shards per table, CrateDB will make a
+best guess, based on the assumption that your nodes have two CPUs each.
+
+### 1000 shards per node limit
+
+To avoid _oversharding_, CrateDB by default limits the number of shards per node to
+1_000 as a protection limit. Any operation that would exceed that limit
+leads to an exception.
+For an 8-node cluster, this allows up to 8_000 total shards across all tables.
+Approaching this limit typically indicates a suboptimal sharding strategy rather
+than optimal performance tuning. See also relevant documentation about
+{ref}`table reconfiguration <number-of-shards>` wrt. sharding options.
+
+### Partitions
+
+If you are using {ref}`partitioned tables <crate-reference:partitioned-tables>`,
+note that each partition is clustered into as many shards as you configure
+for the table.
+
+For example, a table with four shards and two partitions will have eight
+shards that can be commonly queried across. But a query that only touches
+one partition will only query across four shards.
+
+How this factors into balancing your shard allocation will depend on the
+types of queries you intend to run.
 
-### Under-allocation is bad
+### Replicas
+
+CrateDB uses replicas for both data durability and query performance. When a
+node goes down, replicas ensure no data is lost. For read operations, CrateDB
+randomly distributes queries across both primary and replica shards, improving
+concurrent read throughput.
+
+Each replica adds to the total shard count in the cluster. By default, CrateDB
+uses the replica setting `0-1` on newly created tables, resulting in twice the
+number of configured shards. The more replicas you add, the higher the
+multiplier (x3, x4, etc.) for capacity planning
+
+See the {ref}`replication reference <crate-reference:replication>` documentation
+for more details.
+
+### Segments
+
+The number of segments within a shard affects query performance because more
+segments have to be visited.
+
```

Comment:
@matriv, @seut: This section is a bit flat. Do you have any suggestions to unflatten it?

Reply:
Maybe you can use: https://cratedb.com/product/features/lucene-engine and https://lucene.apache.org/core/3_0_3/fileformats.html#Segments and then also mention: https://cratedb.com/docs/crate/reference/en/latest/admin/optimization.html#optimization about the number of segments and how it can be optimized.